upgrade/version.info.template

```diff
@@ -47,7 +55,7 @@ VERSION=@@VERSION@@
 MINIMUM_VERSION=5.3.0.0
 
 #
-# The minimum version from which a reboot after upgrade is optional.
+# The minimum product version from which a reboot after upgrade is optional.
```

Comment:
I'm a little skeptical using "product" and "virtualization" interchangeably.. since, to me, the "product" feels more like the culmination of all of our delphix packages (masking, virtualization, zfs, etc).. IMO, nomenclature is important to get right, and the use of "product" here (and elsewhere) is a bit sketchy to me.

If we could come up with a definition for "product" and document that somewhere, then I'd be more OK with it.